### PR TITLE
Invert nginx-opentracing dependency

### PR DESCRIPTION
**Pull request by willgittoes-dd**
Makes a version of nginx-opentracing module that statically links to our OpenTracing implementation.

Previously, to use NGINX tracing a user would install nginx-opentracing and then dynamically load our OpenTracing implementation. This worked and works fine, but for NGINX module certification it is simpler if there is a single Datadog-managed module that can be certified.

We still want to use the nginx-opentracing module underneath, since it's great! And so that any improvements we make to it can be sent upstream. So this PR includes nginx-opentracing as a submodule, then builds a single, statically-linked, all-in-one NGINX module.

We needed to replace/replicate just the one source file that defines the module, all other code directly references nginx-opentracing.

Other changes are to update documentation, make the build happen, work it into the CI.

### CODE REVIEW

**Comment by willgittoes-dd** [comment]
TODO: Make sure tests look at propagation.h:headerWhitelist, to ensure that anyone who changes the headers doesn't forget to update it.

File: README.md [hunk 1]
```diff
@@ -376,11 +367,9 @@ Annotated Datadog config JSON:
 }
 ```
 
-You also need to provide a JSON-formatted text config file that sets options for the Datadog tracing.
-
 ### Tracing Envoy & Istio
 
-Coming soon!
+See: https://docs.datadoghq.com/integrations/envoy/
```
Comment (cgilmour) [inline README.md]: That's for metrics. Correct URL: https://docs.datadoghq.com/tracing/proxies/envoy/
I'd forgotten to go back and update it in the README

File: include/datadog/opentracing.h [hunk 2]
```diff
@@ -63,26 +63,26 @@ class TraceEncoder {
   virtual ~TraceEncoder() {}
 
   // Returns the Datadog Agent endpoint that traces should be sent to.
-  virtual const std::string &path() = 0;
+  virtual const std::string& path() = 0;
```
Comment (cgilmour) [inline include/datadog/opentracing.h]: Did clang-format change the rules again?

File: scripts/release.sh [hunk 3]
```diff
@@ -101,13 +104,20 @@ echo "${ARTIFACT_URLS}" | while read ARTIFACT_URL
...
+for f in ./*; do
+  [ -f "$f" ] && assets+=(-a "$f")
+done
 
 # Create a github release
 PRERELEASE=$([ $IS_PRERELEASE = true ] && echo "-p" || echo "")
 $GOPATH/bin/hub release create $PRERELEASE \
-  -a linux-amd64-libdd_opentracing_plugin.so.gz \
-  -a linux-amd64-libdd_opentracing_plugin.so.gz.asc \
+  "${assets[@]}"
```
Comment (cgilmour) [inline scripts/release.sh]: backslash at the end?

File: README.md (hunk 1 above)
Comment (willgittoes-dd) [reply README.md]: lol oops

File: include/datadog/opentracing.h (hunk 2 above)
Comment (willgittoes-dd) [reply include/datadog/opentracing.h]: I don't even know! I didn't do this on purpose though!

File: scripts/release.sh (hunk 3 above)
Comment (willgittoes-dd) [reply scripts/release.sh]: TRU DAT

**Pull request merged**